controllers/external-accounts.js

```diff
@@ -14,4 +14,24 @@ const addExternalAccountData = async (req, res) => {
   }
 };
 
-module.exports = { addExternalAccountData };
+const getExternalAccountData = async (req, res) => {
+  try {
+    const externalAccountData = await externalAccountsModel.fetchExternalAccountData(req.query, req.params.token);
+    if (externalAccountData.length === 0) {
+      return res.status(404).json({ message: "No data found" });
+    }
+
+    const attributes = externalAccountData[0].attributes;
+
+    if (attributes.expiry && attributes.expiry < Date.now()) {
+      return res.status(498).json({ message: "Token Expired. Please generate it again" });
+    }
+
+    return res.status(200).json({ message: "Data returned successfully", attributes: attributes });
```

Comment:
Can you please explain more on reason for using two if blocks? Can that be done with if and else if block?

Reply:
Two if block are I think more readable instead of a if else block as those are the edge cases being handled